### PR TITLE
431 code groups csv parser

### LINKED ISSUE #431: Discount Code Import: add code groups
### Description
Discount Code Groups were recently added to the API. In order for customers to appropriately import discount codes including the group information, we need to provide the possibility to import groups within the discount code import. 

API documentation: https://dev.commercetools.com/http-api-projects-discountCodes.html#discountcode

FYI @ValeSauer

### PR DESCRIPTION
**Pull request by wizzy25**
#### Summary

Support parsing of (discount-code)[https://dev.commercetools.com/http-api-projects-discountCodes.html#discountcode] groups in CSV for importing

Also contains minor refactorings to make code more readable, removes the redundant `index.js` file, and removes `lodash` dependency (Minor step in "delodashing" this repo 😃)

resolves #431

### CODE REVIEW

**Comment by tdeekens** [comment]
So what's expected here? Should we review everything again? Have all parts been reviewed in chunks? Is there anything that was added?

**Comment by wizzy25** [comment]
@tdeekens I think you're mixing this PR with #325 
Am I right?

File: packages/csv-parser-discount-code/src/main.js [hunk 1]
```diff
@@ -79,16 +85,25 @@ export default class CsvParserDiscountCode {
...
-      return Object.assign(item, { cartDiscounts })
-    }
-    return item
+    const { cartDiscounts, ...rest } = item
+
+    return cartDiscounts
+      ? Object.assign(rest, {
+          cartDiscounts: cartDiscounts
+            .split(this.multiValueDelimiter)
+            .map(cartDiscount => ({
```
Comment (adnasa) [inline packages/csv-parser-discount-code/src/main.js]: cartDiscount -> cartDiscountId

File: packages/csv-parser-discount-code/src/main.js [hunk 2]
```diff
@@ -40,14 +40,19 @@ export default class CsvParserDiscountCode {
...
 
     this._rowIndex = 0
   }
 
   // Remove fields with empty values from the code objects
   static _removeEmptyFields(item: Object) {
-    return _.omitBy(item, val => val === '')
+    const fileredObj = {}
+    Object.keys(item).forEach(key => {
+      if (item[key] !== '') fileredObj[key] = item[key]
```
Comment (adnasa) [inline packages/csv-parser-discount-code/src/main.js]: use `reduce` 

*cough* transducers is fully possible here *cough*

File: packages/csv-parser-discount-code/src/main.js (hunk 1 above)
Comment (wizzy25) [reply packages/csv-parser-discount-code/src/main.js]: this wouldn't work as I'm returning the object

Comment (adnasa) [reply packages/csv-parser-discount-code/src/main.js]: the point is that you should rename this.
you do assign `cartDiscount` as id in the code below.
when I read `cartDiscount`, I think of the entity, not the id

File: packages/csv-parser-discount-code/src/main.js (hunk 2 above)
Comment (wizzy25) [reply packages/csv-parser-discount-code/src/main.js]: Using `acc` here is the only way to keep lint happy with regards to [this comment](https://github.com/commercetools/nodejs/pull/468#discussion_r169278076)

**Pull request merged**